python/dcr_template.py

```diff
@@ -447,22 +483,23 @@ def _calc_psf_model(self):
         dcr_shift = []
         p0 = self.psf_size//2 - self.kernel_size//2
         p1 = p0 + self.kernel_size
-        for _img, calexp in enumerate(self.exposures):
+        for _img, exp in enumerate(self.exposures):
             el = self.elevation_arr[_img]
             az = self.azimuth_arr[_img]
 
+            # Use the measured PSF as the solution of the shifted PSFs.
+            # Taken at zenith, since we're solving for the shift and don't want to introduce any extra.
             dcr_genZ = _dcr_generator(self.bandpass, pixel_scale=self.pixel_scale, elevation=90., azimuth=az)
+            psf_mat.append(_calc_psf_kernel_full(exp, dcr_genZ, return_matrix=False,
+                           x_size=self.psf_size, y_size=self.psf_size))
+            # Calculate the expected shift (with no psf) due to DCR
             dcr_gen = _dcr_generator(self.bandpass, pixel_scale=self.pixel_scale, elevation=el, azimuth=az)
-            psf_mat.append(_calc_psf_kernel_full(calexp, dcr_genZ, fft=False, reverse_offset=True,
-                           x_size=self.psf_size, y_size=self.psf_size, return_matrix=False))
-            dcr_shift.append(_calc_offset_phase(calexp, dcr_gen, fft=False, reverse_offset=True,
-                             x_size=self.psf_size, y_size=self.psf_size, return_matrix=True))
-        self.debug_psf_image = psf_mat
-        self.debug_dcr_shift = dcr_shift
+            dcr_shift.append(_calc_offset_phase(exp, dcr_gen, return_matrix=True,
+                             x_size=self.psf_size, y_size=self.psf_size))
         psf_mat = np.sum(np.hstack(psf_mat), axis=0)
         dcr_shift = np.hstack(dcr_shift)
-        regularize_psf = self._build_regularization(x_size=self.psf_size,
-                                                    y_size=self.psf_size, n_step=self.n_step)
+        regularize_psf = self._build_regularization(x_size=self.psf_size, y_size=self.psf_size,
+                                                    n_step=self.n_step, frequency_regularization=True)
         regularize_dim = regularize_psf.shape
         vals_use = np.append(psf_mat, np.zeros(regularize_dim[1]))
         kernel_use = np.append(dcr_shift.T, regularize_psf.T, axis=0)
```

Comment:
Not shown here, but in the actual code, shortly after this you have:

```
psf_model = np.reshape(psf_soln[0], (self.n_step, self.psf_size, self.psf_size))
self.psf_model = psf_model[:, p0: p1, p0: p1]
```

It really helps code clarity to avoid using the same name for two different objects. In this case `psf_model` means first one thing, and then something different (though similar). It looks to me as if both might be 2d (?) arrays, but the second version is, well, I'm not sure what `[:, p0: p1, p0: p1]` does and a code comment would help.